### PR TITLE
Added Telescope, a w3m-like gemini client

### PR DESCRIPTION
**Pull request by heph2**
###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Relase notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Review by kira64xyz — CHANGES_REQUESTED** [review]
There is no reference to this derivation, so it never gets called.
Add it to `pkgs/top-level/all-packages`, for example like this:
`telescope = callPackage ../applications/networking/browsers/telescope { }; `

For new packages, you need to set a maintainer. Add yourself to `maintainers/maintainers-list.nix` in a separate commit, formatted as `maintainers: add <your name>`. Afterwards, you can add the `meta.maintainers` attribute to your derivation. See [yafetch](https://github.com/NixOS/nixpkgs/blob/master/pkgs/tools/misc/yafetch/default.nix#L27) for reference, for example.

nixpkgs also has some fairly strict guidelines regarding commits, you should squash them into 1 formatted as `pkg: init at <version>`. Do note that the maintainer list addition would need a seperate commit as well.

If you have any further questions, feel free to ask :heart:

File: pkgs/applications/networking/browsers/telescope/default.nix [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+, bison
+, libevent
+, libressl
+, ncurses
+}:
+
+
+stdenv.mkDerivation rec {
+  pname = "telescope";
+  version = "0.3";
```
Comment (IvarWithoutBones) [inline pkgs/applications/networking/browsers/telescope/default.nix:15]: Is there any reason we are not using `0.3.1`? Seems that just got released the other day, would be great if we could upgrade to that.

File: pkgs/applications/networking/browsers/telescope/default.nix [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+
+  src = fetchFromGitHub {
+    owner = "omar-polo";
+    repo = "telescope";
+    rev = "87e3e8018ef2858201ca3641eff06e130ea3ccd9";
```
Comment (IvarWithoutBones) [inline pkgs/applications/networking/browsers/telescope/default.nix:20]: It looks like you're not fetching the release specified in the `version` attribute, but instead an unstable commit.
```suggestion
    rev = version;
```
When using this, `fetchFromGitHub` will fetch the version specified.

**Review by IvarWithoutBones — CHANGES_REQUESTED** [review]
Just a few comments, let me know if you have any questions :)

**Comment by heph2** [comment]
> Just a few comments, let me know if you have any questions :)

**Comment by heph2** [comment]
> There is no reference to this derivation, so it never gets called.
> Add it to `pkgs/top-level/all-packages`, for example like this:
> `telescope = callPackage ../applications/networking/browsers/telescope { }; `
> 
> For new packages, you need to set a maintainer. Add yourself to `maintainers/maintainers-list.nix` in a separate commit, formatted as `maintainers: add <your name>`. Afterwards, you can add the `meta.maintainers` attribute to your derivation. See [yafetch](https://github.com/NixOS/nixpkgs/blob/master/pkgs/tools/misc/yafetch/default.nix#L27) for reference, for example.
> 
> nixpkgs also has some fairly strict guidelines regarding commits, you should squash them into 1 formatted as `pkg: init at <version>`. Do note that the maintainer list addition would need a seperate commit as well.
> 
> If you have any further questions, feel free to ask heart

That was really helpful. Thanks for the welcome... I'm new to all this ( github etc etc ), so excuse me in advance.
So if i understand correctly, i should rebase and commit with 'pkg: init at <version>' , add my self as maintainer and it should be fine right?

File: pkgs/applications/networking/browsers/telescope/default.nix [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+, pkg-config
+, autoreconfHook
+, bison
+, libevent
+, libressl
+, ncurses
+}:
+
+
+stdenv.mkDerivation rec {
```
Comment (SuperSandro2000) [inline pkgs/applications/networking/browsers/telescope/default.nix:13]:
```suggestion
}:

stdenv.mkDerivation rec {
```

File: pkgs/applications/networking/browsers/telescope/default.nix [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+
+  buildInputs = [
+    libevent
+    libressl
+    ncurses
+  ];
+
+  meta = with lib; {
+    description = "Telescope is a w3m-like browser for Gemini";
+    homepage = "telescope.omarpolo.com";
```
Comment (SuperSandro2000) [inline pkgs/applications/networking/browsers/telescope/default.nix:38]:
```suggestion
    homepage = "https://telescope.omarpolo.com/";
```

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

File: pkgs/applications/networking/browsers/telescope/default.nix (hunk 1 above)
Comment (heph2) [reply pkgs/applications/networking/browsers/telescope/default.nix]: Uh, i'm having troubles with fetchFromGithub and cannot fetch the newest version.
With fetchurl works fine :D

File: pkgs/applications/networking/browsers/telescope/default.nix (hunk 4 above)
Comment (heph2) [reply pkgs/applications/networking/browsers/telescope/default.nix]: Thanks for the suggestions!

**Comment by heph2** [comment]
I've fixed the commits and changed from `fetchFromGitHub` to `fetchurl` ( for fetch 0.3.1 version ).
Thanks to all for the suggestions!

**Pull request merged**